actions/form/class.UserSettings.php

```diff
@@ -114,4 +132,48 @@ private function addTimezoneEl($form): void
             $form->addElement($tzElement);
         }
     }
+
+    private function getUserTimezoneService(): UserTimezoneServiceInterface
+    {
+        if (!$this->userTimezoneService) {
+            $this->userTimezoneService = $this->getServiceManager()->get(
+                UserTimezoneServiceInterface::SERVICE_ID
+            );
+        }
+
+        return $this->userTimezoneService;
+    }
+
+    public function setUserTimezoneService(UserTimezoneServiceInterface $userTimezoneService): void
+    {
+        $this->userTimezoneService = $userTimezoneService;
+    }
+
+    private function getLanguageService(): tao_models_classes_LanguageService
+    {
+        if (!$this->languageService) {
+            $this->languageService = tao_models_classes_LanguageService::singleton();
+        }
+
+        return $this->languageService;
+    }
+
+    public function setLanguageService(tao_models_classes_LanguageService $languageService): void
+    {
+        $this->languageService = $languageService;
+    }
+
+    private function getServiceManager(): ServiceManager
+    {
+        if (!$this->serviceManager) {
+            $this->serviceManager = ServiceManager::getServiceManager();
+        }
+
+        return $this->serviceManager;
+    }
```

Comment:
Lets try to use the DI container instead: 

```suggestion
    private function getContainer(): ContainerInterface
    {
        if (!$this->container) {
            $this->container = ServiceManager::getServiceManager()->getContainer();
        }

        return $this->container;
    }
```

Reply:
Changed in f0d12cbfbb341c933c603a82e884395f14a84f73, thanks!